automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/BatchNodepoolCreationSpec.scala

```diff
@@ -110,12 +110,13 @@ class BatchNodepoolCreationSpec
           _ <- loggerIO.info("About to get app1")
 
           getApp1 = LeonardoApiClient.getApp(googleProject, appName1)
-          monitorApp1CreationResult <- testTimer.sleep(30 seconds) >> streamFUntilDone(getApp1, 60, 10 seconds)(
+          monitorApp1CreationResult <- testTimer.sleep(30 seconds) >> streamFUntilDone(getApp1, 90, 10 seconds)(
             testTimer,
             appDoneCheckable
           ).compile.lastOrError
 
           _ <- loggerIO.info(s"app1 monitor result: ${monitorApp1CreationResult}")
+          _ = monitorApp1CreationResult.status shouldBe AppStatus.Running
```

Comment:
Important note about `streamFUntilDone` -- it doesn't fail if the condition is not met after the timeout! So it's totally possible for the app to still be `Provisioning` after `streamFUntilDone` terminates. So we should always assert the expected state at the end.

Reply:
ahh, yeah I printed it to verify in testing but good to assert, thanks for fixing this I was about to make a PR

Reply:
Yeah, I was printing it to verify during testing but good to assert it as well. Thanks for fixing this, I was about to PR

Reply:
Thanks for this. I didn't quite understand why I was hitting errors when it didn't fully transition on my PR

Reply:
A quick question, I've noticed that we can't do the same sort of status assertion when we use streamFUntilDone with listApps. Is there anything we should do here to make sure the app is finished deleting properly?

Reply:
`streamFUntilDone` is pretty generic -- it just takes an operation and a predicate and polls the operation until the predicate returns true (or a timeout occurs). The important behavior to be aware of is the timeout doesn't cause an exception to occur -- it just continues on. So if you _don't_ want to continue if the predicate didn't become true, you have to add an extra assertion.

So I think `streamFUntilDone` should be able to wait for deletion with listApps -- you just need to define the predicates correctly.